core/collector.class.inc.php

```diff
@@ -265,16 +265,17 @@ protected function GetCharset()
 	protected function GetVersionFromModuleFile()
 	{
 		$aFiles = glob(APPROOT.'collectors/module.*.php');
-		$sModuleFile = null;
-		$sModuleFile = reset($aFiles);
-		if ($sModuleFile == null)
+		if (!$aFiles)
 		{
 			// No module found, use a default value...
 			$this->sVersion = '1.0.0';
+			Utils::Log(LOG_INFO, "No module file found, using default version value.");
+			return;
```

Comment:
logging is great but we cannot afford to leave the return. lot of custom collector developpers forget to provide a module file. their own collector would not work anymore after migrating collector-base.

Reply:
This PR exactly aims to fix this situation. For those collectors, this throws a php warning every time, after merge, you will only get this INFO log message.

Reply:
logging is good to me. only 'return' bothers me. 

when no module file provided, collector will stop its usual processing (earling failure). which can be considered as a regression for our community.

Reply:
It stops the method `GetVersionFromModuleFile` which is exactly what you want, because there is no module file. Line 271 sets the version already. There simply isn't anything else to do in this method.

Reply:
ok agreed.